packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -601,7 +601,7 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
   };
 
   const needsHorizontalScrollbar =
-    containerDimensions.width && columnsTotalWidth > containerDimensions.width;
+    containerDimensions.width && columnsTotalWidth >= containerDimensions.width;
```

Comment:
This was a hidden issue that just wasn't exposed before changes in https://github.com/mui/mui-x/pull/7001
There's no need for a scrollbar if the widths match.